mycore-mods/src/main/java/org/mycore/mods/MCRMODSJobDistributeMetadataJobAction.java

```diff
@@ -0,0 +1,169 @@
+package org.mycore.mods;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jdom2.Content;
+import org.jdom2.Element;
+import org.jdom2.filter.Filter;
+import org.jdom2.filter.Filters;
+import org.mycore.access.MCRAccessException;
+import org.mycore.common.MCRConstants;
+import org.mycore.common.MCRException;
+import org.mycore.common.MCRPersistenceException;
+import org.mycore.common.MCRSession;
+import org.mycore.common.MCRSessionMgr;
+import org.mycore.common.MCRSystemUserInformation;
+import org.mycore.datamodel.common.MCRLinkTableManager;
+import org.mycore.datamodel.metadata.MCRMetaLinkID;
+import org.mycore.datamodel.metadata.MCRMetadataManager;
+import org.mycore.datamodel.metadata.MCRObject;
+import org.mycore.datamodel.metadata.MCRObjectID;
+import org.mycore.frontend.support.MCRObjectIDLockTable;
+import org.mycore.services.queuedjob.MCRJob;
+import org.mycore.services.queuedjob.MCRJobAction;
+
+public class MCRMODSJobDistributeMetadataJobAction extends MCRJobAction {
+
+    public static final String OBJECT_ID_PARAMETER = "id";
+
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    /**
+     * The constructor of the job action with specific {@link MCRJob}.
+     *
+     * @param job the job holding the parameters for the action
+     */
+    public MCRMODSJobDistributeMetadataJobAction(MCRJob job) {
+        super(job);
+    }
+
+    public String getID() {
+        return job.getParameter(OBJECT_ID_PARAMETER);
+    }
+
+    @Override
+    public boolean isActivated() {
+        return true;
+    }
+
+    @Override
+    public String name() {
+        return "Distribute Metadata of " + getID();
+    }
+
+    @Override
+    public void execute() throws ExecutionException {
+        MCRSession session = MCRSessionMgr.getCurrentSession();
+        try {
+            session.setUserInformation(MCRSystemUserInformation.getGuestInstance());
+            session.setUserInformation(MCRSystemUserInformation.getJanitorInstance());
+            MCRObjectID id = MCRObjectID.getInstance(getID());
```

Comment:
Starting from here, there are massive amounts of code duplication between this new class and the existing class `MCRMODSMetadataShareAgent`. This should be avoided, because future update (bug fixes and improvements) might (but in reallity: **will**) not be made consistently in both places.

This class should only be an alternative way to orchestrate the metadata distribution, but should not re-implement the actual logic. [MCR-2895](https://github.com/MyCoRe-Org/mycore/pull/1866) made a better job in this regard (pun intended).